compiler/rustc_lint/src/types.rs

```diff
@@ -331,18 +331,23 @@ fn lint_int_literal<'tcx>(
         }
 
         cx.struct_span_lint(OVERFLOWING_LITERALS, e.span, |lint| {
-            lint.build(&format!("literal out of range for `{}`", t.name_str()))
-                .note(&format!(
-                    "the literal `{}` does not fit into the type `{}` whose range is `{}..={}`",
-                    cx.sess()
-                        .source_map()
-                        .span_to_snippet(lit.span)
-                        .expect("must get snippet from literal"),
-                    t.name_str(),
-                    min,
-                    max,
-                ))
-                .emit();
+            let mut err = lint.build(&format!("literal out of range for `{}`", t.name_str()));
+            err.note(&format!(
+                "the literal `{}` does not fit into the type `{}` whose range is `{}..={}`",
+                cx.sess()
+                    .source_map()
+                    .span_to_snippet(lit.span)
+                    .expect("must get snippet from literal"),
+                t.name_str(),
+                min,
+                max,
+            ));
+            if let Some(sugg_ty) =
+                get_type_suggestion(&cx.typeck_results().node_type(e.hir_id), v, negative)
+            {
+                err.help(&format!("consider using `{}` instead", sugg_ty));
```

Comment:
"using `i8`" sounds weird to me, maybe something like
```suggestion
                err.help(&format!("consider changing its type to `{}`", sugg_ty));
```

Reply:
I just copied-and-pasted this from the binary/hexadecimal error. If we change this one, we should probably change the other one too.

Reply:
hmm, yeah I would prefer us to change this as well? Do you have a preference here?

Reply:
What about

```suggestion
                err.help(&format!("consider using the `{}` type instead", sugg_ty));
```

or something like that?

Reply:
```suggestion
                err.help(&format!("consider using the type `{}` instead", sugg_ty));
```

Reply:
There are things like:

```rust
            ty::Char => FfiUnsafe {
                ty,
                reason: "the `char` type has no C equivalent".into(),
                help: Some("consider using `u32` or `libc::wchar_t` instead".into()),
            },
```

and

```rust
            ty::Str => FfiUnsafe {
                ty,
                reason: "string slices have no C equivalent".into(),
                help: Some("consider using `*const u8` and a length instead".into()),
            },
```

but those are for a different error so leaving them be.